File: SherColorWeb/trunk/FMTinter.Net/FMTinter.Tests1.Net/TinterMessageTest.cs
using NUnit.Framework;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using SWDeviceHandler.domain;

namespace FMTinter.Tests1
{
    [TestFixture]
    public class TinterMessageTest
    { 

    [Test]
    public void TestfromJSON()
    {
        String json = @"{'id':'dc62e861-5d47-4df3-9b21-5bef2c707888','messageName':'TinterMessage','command':'PurgeAll','status':0,'javaMessage':'','commandRC':0,'errorNumber':0,'errorSeverity':0,'errorMessage':0,'errorList':null}";
            TinterMessage msg = TinterMessage.fromJSON(json);
            Assert.True(msg.command.Contains("PurgeAll"));
    }
    [Test]
        public void TesttoJSON()
        {
            TinterMessage msg = new TinterMessage();
            Error e = new Error();
            e.num = 12345;
            e.message = "Tinter Malfunction Test";
            Console.WriteLine("New Error Detail: code {0} string {1}", e.num, e.message);
            msg.errorList.Add(e);
            msg.command = "PurgeAll";
            string json = msg.toJSON();
            Console.WriteLine(json);
            Assert.True(json.Contains("Malfunction"));
        }
    }
}

File: SherColorWeb/SWDeviceHandler-common/src/test/java/com/sherwin/shercolor/swdevicehandler/domain/TinterMessageTest.java
package com.sherwin.shercolor.swdevicehandler.domain;

import static org.junit.Assert.*;

import org.junit.Before;
import org.junit.Test;

import com.google.gson.Gson;
import com.sherwin.shercolor.swdevicehandler.domain.Configuration;
import com.sherwin.shercolor.swdevicehandler.domain.TinterMessage;

public class TinterMessageTest {
	TinterMessage msg = new TinterMessage();
	
	@Before
	public void setUp() throws Exception {
	
	msg.setMessageName("TinterMessage");
	String json ="{\"command\":\"Detect\",\"configuration\":{\"colorantSystem\":\"CCE\",\"model\":\"COROB D200\",\"serial\":\"TESTME\",\"port\":\"COM3\",\"canisterLayout\":[{\"pump\":1,\"code\":\"R4\"},{\"pump\":2,\"code\":\"R3\"},{\"pump\":3,\"code\":\"Y1\"},{\"pump\":4,\"code\":\"NA\"},{\"pump\":5,\"code\":\"N1\"},{\"pump\":6,\"code\":\"G2\"},{\"pump\":7,\"code\":\"L1\"},{\"pump\":8,\"code\":\"R2\"},{\"pump\":9,\"code\":\"W1\"},{\"pump\":10,\"code\":\"Y3\"},{\"pump\":11,\"code\":\"B1\"},{\"pump\":12,\"code\":\"NA\"}],\"canisterMap\":{\"map\":{\"1\":\"R4\",\"2\":\"R3\",\"3\":\"Y1\",\"4\":\"NA\",\"5\":\"N1\",\"6\":\"G2\",\"7\":\"L1\",\"8\":\"R2\",\"9\":\"W1\",\"10\":\"Y3\",\"11\":\"B1\",\"12\":\"NA\"},\"codeMap\":{\"R2\":8,\"R3\":2,\"R4\":1,\"NA\":12,\"N1\":5,\"L1\":7,\"Y1\":3,\"W1\":9,\"Y3\":10,\"G2\":6,\"B1\":11}}},\"status\":0,\"javaMessage\":\"\",\"commandRC\":4,\"errorNumber\":4,\"errorSeverity\":0,\"errorMessage\":\"Command Refused. Tinter Error\",\"lastInitDate\":20171011,\"id\":\"dcfc7428-de48-4939-84d4-b1180a3299de\",\"messageName\":\"TinterMessage\"}";
	msg = new Gson().fromJson(json,TinterMessage.class);

	}
	
	@Test
	public void testSaveToDisk() {
		
		String error = msg.SaveToDisk("test");
		assertEquals("Success",error);
	}
	@Test
	public void testReadFromDisk(){
		String error = msg.SaveToDisk("test"); // junits execute in an arbitrary order so..in case the read executes first
		TinterMessage new_tm = TinterMessage.ReadFromDisk("test");
		assertEquals("Success",error);
		assertNotNull(new_tm);
		assertEquals(new_tm.getLastInitDate(), msg.getLastInitDate());
	}

}
